modules/k8s_eks_addons/ingress-nginx.tf

```diff
@@ -18,9 +18,12 @@ resource "helm_release" "ingress_nginx" {
   dependency_update = true
   values = [
     templatefile("${path.module}/templates/nginx_values.yaml", {
-      public_subnets = join(", ", var.ingress_nginx_config.subnets_ids)
+      public_subnets         = join(", ", var.ingress_nginx_config.subnets_ids)
+      protocol               = var.aws_load_balancer_controller_config.enable ? "ssl" : "tcp"
+      aws_load_balancer_type = var.aws_load_balancer_controller_config.enable ? "external" : "nlb"
     }),
-    var.ingress_nginx_config.chart_values
+    var.aws_load_balancer_controller_config.enable ? var.aws_load_balancer_controller_config.chart_values : var.ingress_nginx_config.chart_values
```

Comment:
Why should the values for the aws-load-balancer-controller helm-chart be used for the ingress-nginx helm-chart?

Reply:
The idea here is to pass the additional annotation to be added to the nginx helm chart, when the aws load balancer controller is enabled. So I had defined these additional annotations in the variable aws_load_balancer_controller_config. But I now realize that this could be misleading and hence I have now added this additional configuration to the ingress_nginx_config itself as an additional field aws_lbc_chart_values. The annotations are then conditionally passed on to the ingress-nginx helm-chart via a ternary operation based on where the aws load balancer controller is enabled or not. Will this a better implementation?